### PR TITLE
ReviewApplication endpoint for fulfilling or denying applications

### PR DESCRIPTION
**Pull request by andresuribe87**
# Overview
The existing SubmitApplication endpoint was updated so it returns an Operation that is not done. In order to mark the operation as done, it needs to be cancelled, or the application needs to be reviewed. This PR does the latter. Note that when an application is reviewed, the application's status is updated and a response is also recorded in the DB. 

# How Has This Been Tested?
As this doesn't introduce new functionality, the existing tests suffice.

# Tracking
OSE-203

### CODE REVIEW

File: pkg/service/manifest/response.go [hunk 1]
```diff
@@ -80,8 +81,14 @@ func (s Service) buildCredentialResponse(applicantDID, manifestID, applicationID
 	}
 
 	// set the information for the fulfilled credentials in the response
-	if err := responseBuilder.SetFulfillment(descriptors); err != nil {
-		return nil, nil, util.LoggingErrorMsg(err, "could not fulfill credential application: could not set fulfillment")
+	if approved {
+		if err := responseBuilder.SetFulfillment(descriptors); err != nil {
+			return nil, nil, util.LoggingErrorMsg(err, "could not fulfill credential application: could not set fulfillment")
+		}
+	} else {
```
Comment (decentralgabe) [inline pkg/service/manifest/response.go:88]: else is unnecessary given the return?

File: pkg/service/manifest/storage/storage.go [hunk 2]
```diff
@@ -237,3 +238,54 @@ func (ms *Storage) DeleteResponse(id string) error {
...
+//     creates in step 2.
+//
+// The operation and it's response (from 3) are returned.
+func (ms *Storage) ReviewApplication(id string, approved bool, reason string, opID string, response StoredResponse) (*StoredResponse, *opstorage.StoredOperation, error) {
+	// TODO: everything should be in a single Tx.
+	m := map[string]any{
+		"status": opsubmission.StatusDenied,
+		"reason": reason,
+	}
+	if approved {
```
Comment (decentralgabe) [inline pkg/service/manifest/storage/storage.go:255]: concurreny access risk?

**Review by thereisnogabe — APPROVED** [review]

**Comment by decentralgabe** [comment]
noticed the e2e test failing - is that new? cc: @nitro-neal

File: pkg/service/manifest/response.go (hunk 1 above)
Comment (andresuribe87) [reply pkg/service/manifest/response.go:88]: It's necessary. Only on of  `fulfillment` or `denial` can be set.

File: pkg/service/manifest/storage/storage.go (hunk 2 above)
Comment (andresuribe87) [reply pkg/service/manifest/storage/storage.go:255]: Yes! This I'm leaving that for a separate PR. I wanted to get a simple implementation out, and then fix the concurrent access issues.

**Review by andresuribe87 — COMMENTED** [review]
> noticed the e2e test failing - is that new? cc: @nitro-neal

Failure was introduced in this PR. Fixed it now. Yay for tests!

File: integration/didweb_integration_test.go [hunk 3]
```diff
@@ -118,7 +119,7 @@ func TestDidWebCreateCredentialManifestIntegration(t *testing.T) {
 	assert.NotEmpty(t, manifestID)
 }
 
-func TestDidWebSubmitApplicationIntegration(t *testing.T) {
+func TestDidWebSubmitAndReviewApplicationIntegration(t *testing.T) {
```
Comment (decentralgabe) [inline integration/didweb_integration_test.go:122]: for the next one `TestDIDWeb`... I know you didn't introduce this

**Pull request merged**